internal/tenant/index.go

```diff
@@ -0,0 +1,110 @@
+package tenant
+
+import (
+	"context"
+	"iter"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	proto "github.com/sourcegraph/zoekt/cmd/zoekt-sourcegraph-indexserver/protos/sourcegraph/zoekt/configuration/v1"
+	"github.com/sourcegraph/zoekt/internal/tenant/internal/tenanttype"
+)
+
+const TenantsDir = "tenants"
+
+// ContextIndexDir returns a context and index dir for the given tenant ID.
+//
+// 🚨 SECURITY: Do not use this function anywhere else than in
+// sourcegraph-indexserver to derive a context and index directory from
+// IndexOptions.
+func ContextIndexDir(tenantID int, repoDir string) (context.Context, string, error) {
+	if !EnforceTenant() {
+		// Default to tenant 1 if enforcement is disabled.
+		tnt, err := tenanttype.FromID(1)
+		if err != nil {
+			return nil, "", err
+		}
+		return tenanttype.WithTenant(context.Background(), tnt), repoDir, nil
+	}
+	tnt, err := tenanttype.FromID(tenantID)
+	if err != nil {
+		return nil, "", err
+	}
+	return tenanttype.WithTenant(context.Background(), tnt), filepath.Join(repoDir, TenantsDir, strconv.Itoa(tenantID)), nil
+}
+
+// HttpExtraHeader returns header we send to gitserver given a tenant context.
+func HttpExtraHeader(ctx context.Context) string {
+	key := headerKeyTenantID + ": "
+	if !EnforceTenant() {
+		return key + "1"
+	}
+	tnt, err := tenanttype.FromContext(ctx)
+	if err != nil {
+		return key + headerValueNoTenant
+	}
+	return key + strconv.Itoa(tnt.ID())
+}
+
+// ListIndexDirs returns a list of index directories for all tenants. If tenant
+// enforcement is disabled, the list is []string{indexDir}.
+func ListIndexDirs(indexDir string) []string {
```

Comment:
this seems very fragile since if any non tenant dir appears in here other places may fail. Maybe each tenant dir should rather have something like a `tenant-` prefix?

Often glob is used underneath. I wonder if instead we just always do `tenant-*/*.zoekt` would be appropriate?